src/compas_fab/robots/configuration.py

```diff
@@ -350,6 +350,78 @@ def close_to(self, other, tol=1e-3):
                 return False
         return True
 
+    def has_joint_names(self):
+        """Returns ``True`` when there is a joint name for every value."""
+        return len(self.values) == len(self.joint_names)
+
+    def check_joint_names(self):
+        """Raises an error if there is not a joint name for every value."""
+        if not self.has_joint_names:
```

Comment:
Sorry, I don't follow...

Reply:
`has_joint_names` is a method, not a property, so, this is never evaluated, it's always true because it checks for method itself not being null (maybe add a failing test case to confirm), so, I think this should either be `if not self.has_joint_names():` or you should add a `@property` decorator to `has_joint_names` definition right above...